pkg/build/gobuild_test.go

```diff
@@ -26,39 +27,110 @@ import (
 	"github.com/google/go-containerregistry/pkg/v1/random"
 )
 
-func TestGoBuildIsSupportedRef(t *testing.T) {
-	base, err := random.Image(1024, 3)
+func createModuleProject(t *testing.T) string {
+	tmpDir, err := ioutil.TempDir("", "module-test")
 	if err != nil {
-		t.Fatalf("random.Image() = %v", err)
+		t.Error(err.Error())
 	}
+	os.Mkdir(filepath.Join(tmpDir, "kodata"), 0777)
+
+	writeFile(t, filepath.Join(tmpDir, "main.go"), `
```

Comment:
Is it possible to pull these out into some checked-in files, and just copy them to a tmpdir? It's kind of ugly with them inlined here. Is this just replicating [these files](https://github.com/google/ko/tree/master/cmd/ko/test)?  You could just copy that directory to `tmpDir` and then write the `go.mod` file (and maybe fix up symlinks? not sure).